src/get-dependencies-from-document.ts

```diff
@@ -8,16 +8,29 @@
  * subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
  */
 
+import {parse as urlParse} from 'url';
 import {DocumentDescriptor} from 'hydrolysis';
 import {posix as posixPath} from 'path';
 import {Node, queryAll, predicates, getAttribute} from 'dom5';
+import * as logging from 'plylog';
+
+const logger = logging.getLogger('cli.build.get-dependencies');
 
 export interface DocumentDeps {
   imports?: Array<string>;
   scripts?: Array<string>;
   styles?: Array<string>;
 }
 
+/**
+ * Detects if a url is remote by checking it's protocol. Also checks if it
+ * starts with '//', which can be an alias to the page's current protocol
+ * in the browser.
+ */
+function isDependencyRemote(url: string) {
+  return urlParse(url).protocol !== null || url.startsWith('//');
```

Comment:
"remote" should include URLs on the same host that are outside the application root, but first we need to add support for application roots. Can you add a TODO here for that?